moml/utils/data_utils/molecular.py

```
@@ -0,0 +1,281 @@
+    if mol_cache_col and mol_cache_col in df.columns:
+        logger.info(f"Using pre-parsed molecules from '{mol_cache_col}' column")
+        df[mol_col] = df[mol_cache_col].apply(lambda mol: Chem.AddHs(mol) if mol is not None else None)
+    else:
+        # Create RDKit molecules and validity flags
+        def smiles_to_mol_with_hs(s):
+            if pd.notna(s) and isinstance(s, str):
+                try:
+                    mol = Chem.MolFromSmiles(str(s))
+                    if mol:
+                        return Chem.AddHs(mol)
+                except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)